### PR TITLE
[Distributed] Fix sparse adagrad

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
The sparse adagrad implementation is different from Pytorch. This leads to lower performance in trained models. This PR is to make sparse adagrad behave the same as Pytorch.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/sparse_emb.py [hunk 1]
```diff
@@ -136,6 +139,15 @@ def step(self):
                     # We'll need to scale them with the state sum on the kvstore server
                     # after we push them.
                     grads = F.cat(grads, 0)
-                    kvstore.push(name, idxs, grads)
+
+                uniq_idxs, inverse_idxs = F.unique(idxs, return_inverse=True)
```
Comment (classicsong) [inline python/dgl/distributed/sparse_emb.py:143]: If it is a distributed system, shall we gather grad from all machines?

Comment (zheng-da) [reply python/dgl/distributed/sparse_emb.py:143]: this requires significant modification. let's have this for now. we need to test if it's necessary to merge all gradients first.
